contributing.md

```diff
@@ -1,115 +1,52 @@
 
-### CONTRIBUTOR LICENSE AGREEMENT
+# Releases
+All the previous releases can be found [here](https://github.com/AzureAD/passport-azure-ad/releases).
 
-Please visit [https://cla.microsoft.com/](https://cla.microsoft.com/) and sign the Contributor License
-Agreement.  You only need to do that once. We can not look at your code until you've submitted this request.
+# Filing Bugs
+Please file issues you see in the [issue tracker](https://github.com/AzureAD/passport-azure-ad/issues). Include:
 
+- The version you are using.
+- The behavior you are seeing. If at all possible, please submit a reduced repro or test that demonstrates the issue.
+- What you expect to see.
 
-### FORK
+# Instructions for Contributing Code
 
-Fork the project [on GitHub][] and check out
-your copy.
+## Contributing bug fixes
 
-Now decide if you want your feature or bug fix to go into the dev branch
-or the master branch.  **All bug fixes and new features should go into the dev branch.**
+We are currently accepting contributions in the form of bug fixes. A bug must have an issue tracking it in the issue tracker. Your pull request should include a link to the bug that you are fixing. If you've submitted a PR for a bug, please post a comment in the bug to avoid duplication of effort.
 
-The master branch is effectively frozen; patches that change the SDKs
-protocols or API surface area or affect the run-time behavior of the SDK will be rejected.
+## Contributing features
+Features (things that add new or improved functionality) may be accepted, but will need to first be approved (tagged with "Enhancement") in the issue.
 
-Some of our SDKs have bundled dependencies that are not part of the project proper.  Any changes to files in those directories or its subdirectories should be sent to their respective
-projects.  Do not send your patch to us, we cannot accept it.
+## Legal
+You will need to complete a Contributor License Agreement (CLA). Briefly, this agreement testifies that you are granting us permission to use the submitted change according to the terms of the project's license, and that the work being submitted is under appropriate copyright.
 
-In case of doubt, open an issue in the [issue tracker][].
+Please submit a Contributor License Agreement (CLA) before submitting a pull request. You may visit https://cla.microsoft.com to sign digitally. You only need to do this once. Once we have received the signed CLA, we'll review the request.
 
-Especially do so if you plan to work on a major change in functionality.  Nothing is more
-frustrating than seeing your hard work go to waste because your vision
-does not align with our goals for the SDK.
+## Housekeeping
+Your pull request should:
 
+* Include a description of what your change intends to do
+* Be based on a reasonably recent pull in the correct branch
+    * Please rebase and squash all commits into a single one
+* Pass all tests
```

Comment:
Looking again, since we have Travis tests running, this should say that you should do your own test run, and then after the PR is submitted, ensure that the automatic Travis test run is passing all tests.